### PR TITLE
[PX-175] CI overview

### PR DESCRIPTION
**Pull request by kylemcnair**
This is an attempt to give a quick overview and definition of CI that will precede the more detailed guides.

### CODE REVIEW

**Comment by kylemcnair** [comment]
Visuals might be an approachable way to explain CI.

Some combination of the following might be nice.

From this Datafold post:
Screen Shot 2022-10-11 at 2 11 15 PM

Screen Shot 2022-10-11 at 2 11 19 PM

Also found this image online: https://code-maze.com/what-is-continuous-integration/
Screen Shot 2022-10-11 at 2 12 41 PM

File: docs/guides/ci_guides_overview.md [hunk 1]
```diff
@@ -3,4 +3,19 @@ sidebar_position: 1
...
+* testing is done manually, if at all.
+* code changes are released at a slower cadence, and with higher rates of failure
+
+A mature CI pipeline:
+* enables scaling teams and codebases
+* offers high-confidence automated test coverage
+* can boost the quantity and quality of developer output
+
+There are many different possible permutations of CI. This guide offers walk-throughs and examples using a few of the most popular tools. 
+
```
Comment (dlawin) [inline docs/guides/ci_guides_overview.md:21]: Having some nav links makes sense here imo 

Ex:
```suggestion
* [dbt Cloud](ci_guides/dbt_cloud.md)
* [dbt Core](ci_guides/dbt_core.md)
    * [CircleCI](ci_guides/dbt_core/circleci.md)
    * [Github Actions](ci_guides/dbt_core/github_actions.md)
    * [Gitlab CI](ci_guides/dbt_core/gitlab_ci.md)
```

Comment (kylemcnair) [reply docs/guides/ci_guides_overview.md:21]: oi yeah that's a good call

**Comment by KGmajor** [comment]
This lgtm, though, the first image you posted @kylemcnair has a typo in it 😞 `important matric`

**Comment by kylemcnair** [comment]
> This lgtm, though, the first image you posted @kylemcnair has a typo in it 😞 `important matric`

Thanks for catching, I just pinged Matt to get that fixed

File: docs/guides/ci_guides_overview.md [hunk 2]
```diff
@@ -3,4 +3,23 @@ sidebar_position: 1
 title: CI Guides
 ---
 
-dsdf
+Continuous integration, or CI, is a well-established practice in software engineering, and is being increasingly adopted by data teams.
```
Comment (williebsweet) [inline docs/guides/ci_guides_overview.md:6]: Serious question - Do users need to know that it's a well-established software engineering practice?

I have been debating this. Does knowing the history of CI in software engineering benefit them or should we pitch them on its merits? I'm starting to think that using the software engineering narrative is a marketing tactic. That tactic can either resonate or obfuscate.

Also - Dan and I have discussed an ultra-simplistic definition ([link](https://docs.google.com/document/d/1AsTo1Hg1UhZPayNoSDJcltuCgkLcm56JPynVKUvJ0jE/edit#heading=h.3ezx54417ix)) - "Put simply, CI (or Continuous Integration) is a process for building, testing, and deploying changes to your code."

File: docs/guides/ci_guides_overview.md [hunk 3]
```diff
@@ -3,4 +3,23 @@ sidebar_position: 1
...
+
+CI streamlines code changes from multiple contributors into a single project. Tests and checks are handled automatically, and ensure the new code’s correctness before integration. By defining and automating this loop, CI can help improve the speed and quality of code changes. 
+
+In organizations without a CI process:
```
Comment (williebsweet) [inline docs/guides/ci_guides_overview.md:10]: I really like this section - it highlights the "alternatives" to CI and the knock-on effects.

Let's call this out in a section called "Without CI" or something like that

File: docs/guides/ci_guides_overview.md [hunk 4]
```diff
@@ -3,4 +3,23 @@ sidebar_position: 1
 title: CI Guides
 ---
 
-dsdf
+Continuous integration, or CI, is a well-established practice in software engineering, and is being increasingly adopted by data teams.
+
+CI streamlines code changes from multiple contributors into a single project. Tests and checks are handled automatically, and ensure the new code’s correctness before integration. By defining and automating this loop, CI can help improve the speed and quality of code changes.
```
Comment (williebsweet) [inline docs/guides/ci_guides_overview.md:8]: @kylemcnair What are your thoughts on this vs the simple definition above?

File: docs/guides/ci_guides_overview.md [hunk 5]
```diff
@@ -3,4 +3,23 @@ sidebar_position: 1
...
+* changes are manually coordinated, and often become a complex synchronization chore.
+* testing is done manually, if at all.
+* code changes are released at a slower cadence, and with higher rates of failure
+
+A mature CI pipeline:
+* enables scaling teams and codebases
+* offers high-confidence automated test coverage
+* can boost the quantity and quality of developer output
+
+There are many different possible permutations of CI. This guide offers walk-throughs and examples using a few of the most popular tools.
```
Comment (williebsweet) [inline docs/guides/ci_guides_overview.md:20]: What about a hook with a prompt like "Interested in setting up CI or optimizing your current setup? Check out our guided walkthroughs."

File: docs/guides/ci_guides_overview.md [hunk 6]
```diff
@@ -3,4 +3,23 @@ sidebar_position: 1
...
+
+In organizations without a CI process:
+* changes are manually coordinated, and often become a complex synchronization chore.
+* testing is done manually, if at all.
+* code changes are released at a slower cadence, and with higher rates of failure
+
+A mature CI pipeline:
+* enables scaling teams and codebases
+* offers high-confidence automated test coverage
+* can boost the quantity and quality of developer output
```
Comment (williebsweet) [inline docs/guides/ci_guides_overview.md:18]: If a user sees this and doesn't have a scaling team or codebase, would this prevent them from setting up CI?

**Comment by williebsweet** [comment]
@kylemcnair Maybe we could add a Loom of what CI looks like for us.

File: docs/guides/ci_guides_overview.md (hunk 3 above)
Comment (kylemcnair) [reply docs/guides/ci_guides_overview.md:8]: I reformatted these sections to "Without CI" and "With CI", and the bullets correspond to one another (ex, no testing vs automated testing)

File: docs/guides/ci_guides_overview.md (hunk 6 above)
Comment (kylemcnair) [reply docs/guides/ci_guides_overview.md]: I switched this to respond to the "Without CI" section

```
#### Without CI
* changes are manually coordinated, and often become a complex synchronization chore.
```

```
#### With CI
* smoothly manage code changes, and scale as your team and codebase grows.

File: docs/guides/ci_guides_overview.md (hunk 5 above)
Comment (kylemcnair) [reply docs/guides/ci_guides_overview.md]: That's a lot better than what I had.

File: docs/guides/ci_guides_overview.md (hunk 4 above)
Comment (kylemcnair) [reply docs/guides/ci_guides_overview.md]: Adopted the simplified def

File: docs/guides/ci_guides_overview.md [hunk 7]
```diff
@@ -3,4 +3,24 @@ sidebar_position: 1
...
+
+Interested in setting up CI or optimizing your current setup? Check out our guided walkthroughs.
+* [dbt Cloud](ci_guides/dbt_cloud.md)
+* [dbt Core](ci_guides/dbt_core.md)
+    * [CircleCI](ci_guides/dbt_core/circleci.md)
+    * [Github Actions](ci_guides/dbt_core/github_actions.md)
+    * [Gitlab CI](ci_guides/dbt_core/gitlab_ci.md)
+
+#### How we use CI at Datafold:
+<iframe width="640" height="414" src="https://www.loom.com/embed/37fc5fb4c1e640c09b9b470128c85a7b" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
```
Comment (dlawin) [inline docs/guides/ci_guides_overview.md:26]: I can't fullscreen the video for some reason

Comment (dlawin) [reply docs/guides/ci_guides_overview.md:26]:
```
Uncaught (in promise) TypeError: fullscreen error
    at U.toggleFullscreen (player-458c8f318ccf220b8302.js:1:179014)
    at player-458c8f318ccf220b8302.js:1:129193
    at Object.Be (vendor~28da250b-acdef3f6f5ee52f69095.js:2:16486)
    at qe (vendor~28da250b-acdef3f6f5ee52f69095.js:2:16640)
    at vendor~28da250b-acdef3f6f5ee52f69095.js:2:34825
    at xr (vendor~28da250b-acdef3f6f5ee52f69095.js:2:34919)
    at Cr (vendor~28da250b-acdef3f6f5ee52f69095.js:2:35333)
    at vendor~28da250b-acdef3f6f5ee52f69095.js:2:40969
    at De (vendor~28da250b-acdef3f6f5ee52f69095.js:2:116261)
    at vendor~28da250b-acdef3f6f5ee52f69095.js:2:36792
    at Tr (vendor~28da250b-acdef3f6f5ee52f69095.js:2:36822)
    at Jt (vendor~28da250b-acdef3f6f5ee52f69095.js:2:24907)
    at Gt (vendor~28da250b-acdef3f6f5ee52f69095.js:2:24133)
    at t.unstable_runWithPriority (vendor~2a6c62b0-302597e0d98bb9bc04a8.js:2:26064)
    at Bl (vendor~28da250b-acdef3f6f5ee52f69095.js:2:45150)
    at Oe (vendor~28da250b-acdef3f6f5ee52f69095.js:2:116001)
    at Yt (vendor~28da250b-acdef3f6f5ee52f69095.js:2:23926)
    at HTMLElement.u (vendor~47c701b7-247b411832996b7208ff.js:1:35183)
    ```
    
    Not sure if that's helpful or not but it throws this exception when I click the button

Comment (dlawin) [reply docs/guides/ci_guides_overview.md:26]: this could be addressed in a separate ticket

**Review by dlawin — APPROVED** [review]

File: docs/guides/ci_guides_overview.md (hunk 7 above)
Comment (dlawin) [reply docs/guides/ci_guides_overview.md:26]: captured in px-217

**Pull request merged**